File: Algorithm/Python/153. Find Minimum in Rotated Sorted Array.py
# -*- coding: utf-8 -*-
# @Author: WuLC
# @Date:   2016-08-03 16:23:35
# @Last modified by:   WuLC
# @Last Modified time: 2016-08-03 16:23:47
# @Email: liangchaowu5@gmail.com

class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0: return None
        left, right = 0, len(nums)-1
        while left < right:
            mid = (left+right)/2
            if nums[right] < nums[mid]:
                left = mid + 1
            else:
                right = mid
        return nums[left]

File: Algorithm/Java/153. Find Minimum in Rotated Sorted Array.java
/*
* @Author: WuLC
* @Date:   2017-05-27 11:56:13
* @Last Modified by:   WuLC
* @Last Modified time: 2017-05-27 11:56:44
* @Email: liangchaowu5@gmail.com
*/


// binary search, pay attention that one half must be in order
public class Solution 
{
    public int findMin(int[] nums) 
    {
        int left = 0 , right = nums.length - 1, mid;
        while (left < right)
        {
            mid = left + ((right - left) >> 1);
            if (nums[right] > nums[mid]) right = mid;
            else left = mid + 1;
        }
        return nums[left];
    }
}